flang/include/flang/Optimizer/OpenMP/Passes.td

```diff
@@ -50,4 +50,16 @@ def FunctionFilteringPass : Pass<"omp-function-filtering"> {
   ];
 }
 
+def GenericLoopConversionPass
+    : Pass<"omp-generic-loop-conversion", "mlir::func::FuncOp"> {
```

Comment:
Is there any chance we could have loop wrapper ops nested inside of other operations some day?

I'm not saying this is a real case - just an example: say we used a parallel do inside of an omp.private copy region to copy each element of an array in parallel.

I imagine this is a no, I just wanted to check.

Reply:
That's a good quesiton and I am not sure about the answer. But my intuitiion is that even if we need a loop to initialize a `firstprivate` variable, then this loop will exists in a function of its own; for example, a copy constructor or an overload of the assignment operator which means the loop is still inside a function.

@mjklemm please correct me if I am wrong here.

Reply:
I think that's correct.  In that case, the generated code would likely call a runtime function for intrinsic assignment.  Maybe the best way is to clearly document this assumption, so that when it turns out wrong in the future we can easily update it.

Reply:
Done.